lib/Doctrine/Migrations/Tools/Console/Command/MigrateCommand.php

```diff
@@ -121,7 +121,7 @@ protected function execute(InputInterface $input, OutputInterface $output) : int
         $migratorConfigurationFactory = $this->getDependencyFactory()->getConsoleInputMigratorConfigurationFactory();
         $migratorConfiguration        = $migratorConfigurationFactory->getMigratorConfiguration($input);
 
-        $question = 'WARNING! You are about to execute a database migration that could result in schema changes and data loss. Are you sure you wish to continue?';
+        $question = sprintf('WARNING! You are about to execute a migration in database "%s" that could result in schema changes and data loss. Are you sure you wish to continue?', $this->getDependencyFactory()->getConnection()->getDatabase());
```

Comment:
`getDatabase` is not part of the `\Doctrine\DBAL\Driver\Connection` interface, thus i'm not sure if we can do this...

@greg0ire ?

Reply:
:thinking: but `getConnection` returns a `\Doctrine\DBAL\Connection` instead of that interface, right?

Reply:
hmm. right. being so, we know the database... (i remember that there was a reason for choosing the concrete implementation, but can't recall why :-( )

Reply:
In doctrine/migrations **2.2.1** I'm getting:
> In MigrateCommand.php line 151:
> Attempted to call an undefined method named "getDependencyFactory" of class "Doctrine\Bundle\MigrationsBundle\Command\MigrationsMigrateDoctrineCommand". Did you mean to call "setDependencyFactory"?

But in **3.0.1** `get_class($this->getDependencyFactory()->getConnection())` gives me:
> Doctrine\DBAL\Connection

Reply:
the command `getDependencyFactory` has been introduced in 3.0

Reply:
@greg0ire I'm not sure if is worth this, it would be nice to change the return type from `\Doctrine\DBAL\Driver\Connection` to `\Doctrine\DBAL\Connection`..

Reply:
That wouldn't be a BC break, except for extending classes, so… go ahead?

Reply:
May be I'm worrying too much. Probably the improved dx is worth this change.

@ThomasLandauer  can you do the same change on the other commands?

Reply:
Sure, done :-) (I just copied it over - since they all extend `DoctrineCommand`)

@goetas: What means "DX"? Developer Experience - as opposed to User Experience (UX)?

Reply:
@ThomasLandauer yes, DX means Developer Experience